dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -1389,152 +1389,201 @@ public Identifier getRelatedIdentifier(Contentlet contentlet,String relationship
-                Logger.debug(this, errorMessage + ". An empty list will be returned", e);
+            } else if (e instanceof SearchPhaseExecutionException || e
+                    .getCause() instanceof SearchPhaseExecutionException) {
+                Logger.warnAndDebug(ESContentletAPIImpl.class,
+                        errorMessage + ". An empty list will be returned", e);
                 return Collections.emptyList();
             }
             throw new DotDataException(errorMessage, e);
         }
     }
 
-    private String getRelatedContentESQuery(Contentlet contentlet, Relationship rel) {
-        final boolean isSameStructRelationship = FactoryLocator.getRelationshipFactory()
-                .sameParentAndChild(rel);
-        String q;
+    private List<Contentlet> getRelatedChildren(final Contentlet contentlet, final Relationship rel,
+            final User user, final boolean respectFrontendRoles)
+            throws DotSecurityException, DotDataException {
 
-        if(isSameStructRelationship) {
-            q = "+type:content +(" + rel.getRelationTypeValue() + "-parent:" + contentlet.getIdentifier() + " " +
-                    rel.getRelationTypeValue() + "-child:" + contentlet.getIdentifier() + ") ";
-            if(!InodeUtils.isSet(contentlet.getIdentifier())){
-                q = "+type:content +(" + rel.getRelationTypeValue() + "-parent:" + "0 " +
-                        rel.getRelationTypeValue() + "-child:"  + "0 ) ";
-            }
-        } else {
-            q = "+type:content +" + rel.getRelationTypeValue() + ":" + contentlet.getIdentifier();
-            if(!InodeUtils.isSet(contentlet.getIdentifier())){
-                q = "+type:content +" + rel.getRelationTypeValue() + ":" + "0";
+        final List<Contentlet> result = new ArrayList<>();
+        final String relationshipName = rel.getRelationTypeValue().toLowerCase();
+
+        SearchResponse response = APILocator.getEsSearchAPI()
+                .esSearchRelated(contentlet.getIdentifier(), relationshipName, false,false, user,
+                        respectFrontendRoles);
+
+        if (response.getHits() == null) {
+            return result;
+        }
+
+        for (SearchHit sh : response.getHits()) {
+            try {
+                Map<String, Object> sourceMap = sh.getSourceAsMap();
+                if (sourceMap.get(relationshipName) != null) {
+                    ((ArrayList<Map>) sourceMap.get(relationshipName)).stream().forEach(child -> {
+                        try {
+                            result.add(findContentletByIdentifierAnyLanguage(
+                                    (String) child.get("identifier")));
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    });
+                }
+            } catch (Exception e) {
+                Logger.error(this, e.getMessage(), e);
```

Comment:
Please log the content's Identifier and relationship name too.

Reply:
The catch was not necessary, so I removed it